docs/src/tutorials/getting_started/design_patterns_for_larger_models.jl

```diff
@@ -0,0 +1,645 @@
+    config::_AbstractConfiguration,
+)
+    return _solve_knapsack(optimizer, _read_data(data_filename), config)
+end
+
+end
+
+# Finally, you can call your model:
+
+import .KnapsackModel
+
+KnapsackModel.solve_knapsack(
+    GLPK.Optimizer,
+    joinpath(@__DIR__, "data", "knapsack.json"),
+    KnapsackModel.BinaryKnapsack(),
+)
+
+# !!! note
+#     The `.` in `.KnapsackModel` denotes that it is a submodule and not a
+#     separate package that we installed with `Pkg.add`. If you put the
+#     `KnapsackModel` in a separate file, load it with:
+#     ```julia
+#     include("path/to/KnapsackModel.jl")
+#     import .KnapsackModel
+#     ```
+
+# ## Add tests
+
+# As a final step, you should add tests for your model. This often means testing
+# on a small problem for which you can work out the optimal solution by hand.
+# The Julia standard library `Test` has good unit-testing functionality.
+
+import .KnapsackModel
+using Test
+
+@testset "KnapsackModel" begin
+    @testset "feasible_binary_knapsack" begin
+        x = KnapsackModel.solve_knapsack(
+            GLPK.Optimizer,
+            joinpath(@__DIR__, "data", "knapsack.json"),
+            KnapsackModel.BinaryKnapsack(),
+        )
+        @test isapprox(x["apple"], 1, atol = 1e-5)
+        @test isapprox(x["banana"], 0, atol = 1e-5)
+        @test isapprox(x["cherry"], 0, atol = 1e-5)
+        @test isapprox(x["date"], 1, atol = 1e-5)
+        @test isapprox(x["eggplant"], 1, atol = 1e-5)
+    end
+    @testset "feasible_integer_knapsack" begin
+        x = KnapsackModel.solve_knapsack(
+            GLPK.Optimizer,
+            joinpath(@__DIR__, "data", "knapsack.json"),
+            KnapsackModel.IntegerKnapsack(),
+        )
+        @test isapprox(x["apple"], 0, atol = 1e-5)
+        @test isapprox(x["banana"], 0, atol = 1e-5)
+        @test isapprox(x["cherry"], 0, atol = 1e-5)
+        @test isapprox(x["date"], 5, atol = 1e-5)
+        @test isapprox(x["eggplant"], 0, atol = 1e-5)
+    end
+    @testset "feasible_binary_knapsack" begin
+        x = KnapsackModel.solve_knapsack(
+            GLPK.Optimizer,
+            ## This file contains data that makes the problem infeasible.
+            joinpath(@__DIR__, "data", "knapsack_infeasible.json"),
+            KnapsackModel.BinaryKnapsack(),
+        )
+        @test x === nothing
+    end
+end
+
+# !!! tip
+#     Place these tests in a separate file `test_knapsack_model.jl` so that you
+#     can run the tests by adding `include("test_knapsack_model.jl")` to any
+#     file where needed.
+
+# ## Next steps
+
+# We've only briefly scratched the surface of ways to create and structure large
+# JuMP models, so consider this tutorial a starting point, rather than a
+# comprehensive list of all the possible ways to structure JuMP models.  If you
+# are embarking on a large project that uses JuMP, a good next step is to
+# look at ways people have written large JuMP projects "in the wild". Here are
+# some examples:
```

Comment:
NOt sure whether it's bad practice to link to youtube videos, but the [PowerModels-talk](https://youtu.be/POOt1FCA8LI?t=1200) contains some slides that fit the content of the tutorial so neatly that I personally would have found it much more helpful to be taken to that excerpt from the talk rather than the repo (where given the intimidating codebase I might not even attempt to find the design patterns)